### ISSUE TITLE
Get rid of using Matrix “account data” feature

### ISSUE THREAD

**Issue by err508** [raiden-network/raiden#4643]
The transport currently recovers known room_ids after a restart via the account data retrieved from the server. There are two problems with this: 

- it is slow
- it is not supported by the matrix-nio

As discussed in transport weekly (19.8), the plan is to retrieve room_ids after restarts via caching/storage (#4089). Therefore as a first step, room_ids have to be kept track off in the address_mgr.

**Comment by ulope** [comment]
Thinking a bit more about this I'm not sure the UAM is the right place to keep this information.

There are two reasons I currently see to keep the room ids out:
- The UAM is shared with the services, so we should keep the 'pollution' with Raiden client specific features to a minimum.
- Rooms have really not much to do with what it's currently doing.

An option would of course be to introduce something like `PeerRoomManager` (or similar) so the room handling could be extracted out of the transport.

**Comment by err508** [comment]
For me it seems the other way around; The UserAddressManager's job is to _manage_ _address_ specific data for the transport. User_ids are address specific, rooms are user_id specific, therefore are address specific by transitivity. 
Breaking User_id management and room_id management into separate units is therefore a hard to follow abstraction and will lead to hard to comprehend code, especially when the storage then also has to be called in multiple places. 
I agree to your point on the service's coupling though: we could move the service's utilities to a more lightweight parent class.

**Comment by ulope** [comment]
Replaced by #5558
